practice/rain.py

```diff
@@ -0,0 +1,130 @@
+"""Takes data from rain gauge at Sunnyside and prints a summary of the data: date with the most rain &
+year with most rain.
+"""
+# setup:
+filename = 'sunnyside.txt'
+
+
+def each_line_of_file(filename):
+    """Opens sunnyside.txt and reads it line by line."""
+    with open(filename, newline='') as f:
+        lines = f.readlines()
+
+    return lines
+
+
+def split_lines(lines):
+    r"""Strips and splits the lines into lists.
+
+    >>> split_lines(['1 \n', '2\n', '3\n', '4\n', '5\n', '6 \n', '7 \n', '8 \n', '9 \n', '10 \n', '11 \n', 'PARTY! \n', 'TIME!'])
+    [['PARTY!']]
+    """
+    listed_lines = [x.strip().split('\n') for x in lines]
+    listed_lines_reduced = listed_lines[11:-1]
+    line_list = []
+    for listed_lines_split in listed_lines_reduced:
+        line_list += [y.split() for y in listed_lines_split]
```

Comment:
The point of splitting this into two functions is not because the intermediate value is useful everywhere, it's that you need to be able to validate that each step is functioning properly and breaking it into two functions allows that to happen easily.

Arguably the whole program should be one function because the person using the program doesn't care about anything but the final output value... The whole point of program structure is for you and validation.